### PR TITLE
Import firestore conformance tests from googleapis/conformance-tests@32fd66cf7dffe8431643f08b5944108ac10c8bc2

### PR DESCRIPTION
**Pull request by BenWhitehead**
* Add new script generate-conformance-tests.sh that can reliably update
  the various files that make up the conformance tests.
  * Currently the script only updates firestore tests.
  * Updating of the git-submodule linking to conformance-tests is not
    done by the script, it is expected that the developer will do the
    bumping.
* import proto definition
* generate updated TestDefinition from definition
* Add individual test files
* Add maven profile `gen-conformance-protos` that can be used to generate protos in `com.google.cloud:google-cloud-conformance-tests`

Depends on #5540

### CODE REVIEW

**Comment by BenWhitehead** [comment]
@kolea2 Please review

**Comment by BenWhitehead** [comment]
> Are all the languages using submodules for reading the tests?

They are starting to move in that direction. dotnet is already using sub-module, and ruby is going to move when updating the firestore tests.

I will note, the submodule here is purely to track the commit the resources are from, the sub-module does not need to be cloned for the tests to run. The script `generate-conformance-tests.sh` copies and generates everything into the correct location so the tests can run.

**Comment by BenWhitehead** [comment]
The builds are currently failing because I need to rebase them. Since this PR is dependent on another PR I'll wait to rebase until that PR is merged. The changes in the PR are unrelated to the build failures and can still be reviewed.

**Comment by BenWhitehead** [comment]
Rebased after merge of #5540 and ready for review.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "google-cloud-testing/google-cloud-conformance-tests/conformance-tests"]
+	path = google-cloud-testing/google-cloud-conformance-tests/conformance-tests
+	url = git@github.com:googleapis/conformance-tests.git
```
Comment (kolea2) [inline .gitmodules]: Just for my understanding, what does this do?

**Comment by kolea2** [comment]
(can't comment on the file as it is too big) Is TestDefinition.java generated or hand written?

File: .gitmodules (hunk 1 above)
Comment (BenWhitehead) [reply .gitmodules]: This file is the definition for the git-submodule which is being used to track which revision the local conformance test resources came from. You can see an example of what it looks like and how it exists in the file tree [here](https://github.com/BenWhitehead/google-cloud-java/tree/firestore/conformance-tests-migration/step2/google-cloud-testing/google-cloud-conformance-tests).

**Comment by BenWhitehead** [comment]
> (can't comment on the file as it is too big) Is TestDefinition.java generated or hand written?

TestDefinition.java is generated from [`tests.proto`](https://github.com/BenWhitehead/google-cloud-java/blob/firestore/conformance-tests-migration/step2/google-cloud-testing/google-cloud-conformance-tests/src/main/proto/google/cloud/conformance/firestore/v1/tests.proto).

**Comment by kolea2** [comment]
LGTM, @chingor13 do you have any final comments?

**Comment by chingor13** [comment]
LGTM, but can you add a bit more info on the README for the conformance tests on the relationship between the test protos and the json config files?

**Comment by BenWhitehead** [comment]
@chingor13 Added some details to the README to explain the files and their relationships. I've also rebased the PR on master.

**Pull request merged**